spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -315,16 +317,37 @@ private boolean obtainLock() {
 			 */
 			this.lockedAt = System.currentTimeMillis();
 			this.threadName = currentThread.getName();
-			Boolean success = RedisLockRegistry.this.redisTemplate.boundValueOps(
-					constructLockKey()).setIfAbsent(this);
+
+			Boolean success = RedisLockRegistry.this.redisTemplate.execute(new SessionCallback<Boolean>() {
+				@SuppressWarnings("unchecked")
```

Comment:
`@SuppressWarnings({"unchecked", "rawtypes"})` to get rid of compilation warns.
Actually I don't like this `execute(RedisOperations ops)` with that generic interface, because we can't specify them here for inline `SessionCallback` impl. Anyway we don't have a choice and just should suppress those warns.
Will be fixed during merge...
Please, don't do any change any more